src/system_info/system_info_instance.cc

```diff
@@ -67,9 +69,11 @@ void SystemInfoInstance::InstancesMapInitialize() {
   RegisterClass<SysInfoDeviceOrientation>();
   RegisterClass<SysInfoDisplay>();
   RegisterClass<SysInfoLocale>();
-  RegisterClass<SysInfoNetwork>();
   RegisterClass<SysInfoPeripheral>();
-#if defined(TIZEN_IVI) || defined(TIZEN_MOBILE)
+#ifdef GENERIC_DESKTOP
+  RegisterClass<SysInfoNetworkDesktop>();
+#else
+  RegisterClass<SysInfoNetworkTizen>();
```

Comment:
Talked with @halton offline. We reached an agreement to use current design. No need to change.